### PR TITLE
Disable profile controls if the HS doesn't allow them to be set

### PR DESCRIPTION
**Pull request by dbkr**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/4246
Based on https://github.com/matrix-org/matrix-react-sdk/pull/12650

New version of https://github.com/matrix-org/matrix-react-sdk/pull/12634 split out from compound updates.

## Checklist

-   [ ] Tests written for new code (and old code if feasible).
-   [ ] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [ ] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

**Comment by dbkr** [comment]
Aaaargh, sorry for the mess on this PR, I had a complete failure pressing the merge button on a PR stacked on top of this one and it's going downhill from there. I'm just going to force-push to fix it.

File: src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx [hunk 1]
```diff
@@ -167,7 +171,7 @@ export default class GeneralUserSettingsTab extends React.Component<IProps, ISta
     private async getCapabilities(): Promise<void> {
         const cli = this.context.client!;
 
-        const capabilities = await cli.getCapabilities(); // this is cached
+        const capabilities = cli.getCachedCapabilities() ?? {};
```
Comment (robintown) [inline src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx:174]: If I'm understanding this correctly, then if you open user settings before capabilities have been fetched, the controls will stay disabled until you close and re-open settings? `await cli.getCapabilities()` seems preferable in that regard.

Comment (robintown) [reply src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx:174]: I agree a spinner is a bit overkill for this corner case, I was just thinking that allowing the controls to unlock eventually is better than leaving them stuck in an inaccurate state (and the method is already async, so, should be low-effort to switch to `getCapabilities`?)

Comment (dbkr) [reply src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx]: As yes true, the method is async anyway - done.

**Review by robintown — APPROVED** [review]

**Pull request merged**